src/modules/currency/formatAmount.ts

```diff
@@ -0,0 +1,59 @@
+import { CURRENCIES } from "./data/currencies";
+
+// this function formats amount based on locale and options provided
+const formatAmount = (
+  currencyCode: keyof typeof CURRENCIES,
+  amount: string | number,
+  options: {
+    locale?: string;
+    withSymbol?: boolean;
+    currencyDisplay?: "code" | "symbol" | "narrowSymbol" | "name";
+    currencySign?: "standard" | "accounting";
+  } = {}
+): string => {
+  if (!Number(amount)) throw new Error("Parameter `amount` is not a number!");
+
+  const {
+    withSymbol = true,
+    currencyDisplay = "symbol",
+    currencySign = "standard",
+  } = options;
+  let { locale } = options;
+
+  const numberFormatOptions: Intl.NumberFormatOptions = {
+    style: withSymbol ? "currency" : "decimal",
+    currency: currencyCode as string,
+    currencySign: currencySign,
+  };
+
+  // Define a separate options object for currencyDisplay
+  const currencyDisplayOptions: Record<string, string> = {
+    currencyDisplay: currencyDisplay,
+  };
+
+  // Merge the two options objects
+  const mergedOptions = {
```

Comment:
don't see much benefit in keeping 3 variables for this... One might think why is currencySign in `numberFormatOptions` and not in `currencyDisplayOptions`

Better to use a single object for options here.